cmake/ZeekSpicyAnalyzerSupport.cmake

```diff
@@ -6,32 +6,65 @@
 
 include(GNUInstallDirs)
 
-# Add target to build an analyzer. Arguments are the name of the analyzer and a
-# variable number of source files for `spicyz`.
-function(spicy_add_analyzer name)
-    set(sources "${ARGN}")
-    string(TOLOWER "${name}" name_lower)
-    set(output "${SPICY_MODULE_OUTPUT_DIR_BUILD}/${name_lower}.hlto")
-    set(output_install "${SPICY_MODULE_OUTPUT_DIR_INSTALL}/${name_lower}.hlto")
-    set(deps "spicyz")
+# Add target to build an analyzer.
+#
+# Usage:
+#
+#     spicy_add_analyzer(
+#         NAME name
+#         [SOURCES <source files for spicyz>...]
+#         [SCRIPTS <additional script files to install>...]
+#     )
+function(spicy_add_analyzer)
+    set(options)
+    set(oneValueArgs NAME)
+    set(multiValueArgs SOURCES SCRIPTS)
+
+    cmake_parse_arguments(PARSE_ARGV 0 SPICY_ANALYZER "${options}" "${oneValueArgs}" "${multiValueArgs}")
+
+    # We also support the legacy behavior where the first arg is
+    # the analyzer NAME and all remaining arguments are SOURCES.
+    if (SPICY_ANALYZER_UNPARSED_ARGUMENTS)
+        if (SPICY_ANALYZER_NAME OR SPICY_ANALYZER_SOURCES OR SPICY_ANALYZER_SCRIPTS)
+            message(FATAL_ERROR "named an unnamed arguments cannot be mixed")
+        endif()
+
+        list(GET ARGN 0 SPICY_ANALYZER_NAME)
+        list(POP_FRONT ARGN)
+
+        set(SPICY_ANALYZER_SOURCES ${ARGN})
+    endif()
+
+    if (NOT DEFINED SPICY_ANALYZER_NAME)
+        message(FATAL_ERROR "NAME is required")
+    endif()
+
+    string(TOLOWER "${SPICY_ANALYZER_NAME}" NAME_LOWER)
+    set(OUTPUT "${SPICY_MODULE_OUTPUT_DIR_BUILD}/${NAME_LOWER}.hlto")
 
     add_custom_command(
-        OUTPUT ${output}
-        DEPENDS ${sources} ${deps}
-        COMMENT "Compiling ${name} analyzer"
+        OUTPUT ${OUTPUT}
+        DEPENDS ${SPICY_ANALYZER_SOURCES} spicyz
+        COMMENT "Compiling ${SPICY_ANALYZER_NAME} analyzer"
         COMMAND mkdir -p ${SPICY_MODULE_OUTPUT_DIR_BUILD}
-        COMMAND spicyz -o ${output} ${SPICYZ_FLAGS} ${sources}
+        COMMAND spicyz -o ${OUTPUT} ${SPICYZ_FLAGS} ${SPICY_ANALYZER_SOURCES}
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
         )
 
-    add_custom_target(${name} ALL DEPENDS ${output})
+    add_custom_target(${SPICY_ANALYZER_NAME} ALL DEPENDS ${OUTPUT})
 
     if ( SPICY_MODULE_OUTPUT_DIR_INSTALL )
-        install(FILES ${output} DESTINATION "${SPICY_MODULE_OUTPUT_DIR_INSTALL}")
+        install(FILES ${OUTPUT} DESTINATION "${SPICY_MODULE_OUTPUT_DIR_INSTALL}")
+
+        if (DEFINED SPICY_ANALYZER_SCRIPTS)
+            install(
+                FILES ${SPICY_ANALYZER_SCRIPTS}
+                DESTINATION "${SPICY_SCRIPTS_OUTPUT_DIR_INSTALL}/spicy-analyzers/${NAME}")
```

Comment:
This won't quite work yet: In the future we want to use this also for packages other than `spicy-analyzers`, so we shouldn't hardcode the name. Not immediately sure though what the solution is. Do we need another argument `PACKAGE_NAME` for this?

Reply:
Good catch. I went with an explicit `PACKAGE_NAME` argument which becomes required if `SCRIPTS` is also given.